src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2470,6 +2470,24 @@ static bool contextElementPreconditionsCheck
   /* Getting the entity in the request (helpful in other places) */
   EntityId* enP = &ceP->entityId;
 
+  /* Checking there aren't duplicate attributes */
+  for (unsigned int ix = 0; ix < ceP->contextAttributeVector.size(); ++ix)
+  {
+    std::string name = ceP->contextAttributeVector.get(ix)->name;
+    std::string id   = ceP->contextAttributeVector.get(ix)->getId();
+    for (unsigned int jx = ix + 1; jx < ceP->contextAttributeVector.size(); ++jx)
```

Comment:
Of course, this is how you do it:

0: compared to 1-max
1: compared to 2-max  (has already been compared to 0)
2: compared to 3-max (has already been compred to 0 and 1)
etc

NTC